source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -998,6 +998,337 @@ See the `Load Balancer Specification <../load-balancers/load-balancers.rst#event
+
+  Connection pool created for host with hostname {{serverHost}} on port {{serverPort}} using options maxIdleTimeMS={{maxIdleTimeMS}},
+  minPoolSize={{minPoolSize}}, maxPoolSize={{maxPoolSize}}, maxConnecting={{maxConnecting}}, waitQueueTimeoutMS={{waitQueueTimeoutMS}},
+  waitQueueSize={{waitQueueSize}}, waitQueueMultiple={{waitQueueMultiple}}
+
+Pool Ready Message
+------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection pool ready"
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection pool ready for host with hostname {{serverHost}} on port {{serverPort}}
+
+Pool Cleared Message
+--------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection pool cleared"
+
+   * - serviceId
+     - String
+     - The hex string representation of the service ID which the pool was cleared for. Optional; only present in load balanced mode.
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection pool cleared for serviceId {{serviceId}} and host with hostname {{serverHost}} on port {{serverPort}}
+
+Pool Closed Message
+-------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection pool closed"
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection pool closed for host with hostname {{serverHost}} on port {{serverPort}}
+
+Connection Created Message
+--------------------------
+In addition to the common fields defined above, this message MUST contain the following key-value pairs:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Key
+     - Suggested Type
+     - Value
+
+   * - message
+     - String
+     - "Connection created"
+
+   * - driverConnectionId
+     - Int
+     - The driver-generated ID for the connection as defined in `Connection <#connection>`_.
+
+The unstructured form SHOULD be as follows, using the values defined in the structured format above to fill in placeholders as appropriate:
+
+  Connection created with driver ID {{driverConnectionId}} to host with hostname {{serverHost}} on port {{serverPort}}
```

Comment:
"driver ID" here sort of reads like its identifying the driver that created the connection rather than the connection itself. Do you think it would be too ambiguous to just say "ID" here?

Reply:
I'd like to make sure that it's clear this is a driver-generated ID, vs the server-generated IDs that they will see in server logs. I don't think it would necessarily be obvious to a user that at the time this message is emitted we don't have access to the server ID yet. maybe we could just say "driver-generated ID"? 

relatedly: I used "driver ID" in all the command log message suggested unstructured forms. but that could be revised to be consistent with these

Reply:
updated to "driver-generated ID" here and elsewhere. like above, will make similar change to command messages in follow up PR